src/lib.rs

```diff
@@ -279,6 +282,12 @@ impl UnixStream {
         self.io.get_ref().peer_addr()
     }
 
+    /// Returns effective credentials of the process which called `connect` or `socketpair`.
+    #[cfg(any(target_os = "linux", target_os = "macos"))]
+    pub fn peer_cred(&self) -> io::Result<UCred> {
```

Comment:
This either needs to be ungated to cause compile failures on other platforms to necessitate an implementation, or it needs an extension trait. I don't mind which, but there's no middle between those tw.